frontend/app_flowy/packages/appflowy_editor/lib/src/service/internal_key_event_handlers/checkbox_event_handler.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:flutter/material.dart';
+
+ShortcutEventHandler toggleCheckbox = (editorState, event) {
+  final selection = editorState.service.selectionService.currentSelection.value;
+  final nodes = editorState.service.selectionService.currentSelectedNodes;
+  final checkboxTextNodes = nodes
+      .where(
+        (element) =>
+            element is TextNode &&
+            element.subtype == BuiltInAttributeKey.checkbox,
+      )
+      .toList(growable: false);
+
+  if (selection == null || checkboxTextNodes.isEmpty) {
+    return KeyEventResult.ignored;
+  }
+
+  //If any one of the checkboxes is unchecked then make all checkboxes checked
+
+  bool isAllCheckboxesChecked = true;
```

Comment:
```dart
  bool isAllChcked = checkboxTextNodes
      .every((e) => node.attributes[BuiltInAttributeKey.checkbox] == true);
  final transaction = editorState.transaction;
  if (isAllChcked) {
    // if all the checkboxes are checked, then make all of the checkboxes unchecked

    // apply the transaction to make all checkboxes unchecked.
    // ...
  } else {
    // If any one of the checkboxes is unchecked then make all checkboxes checked

    // apply the transaction to make all checkboxes checked.
    // ...
  }
```


No need to use `!` to get the opposite value. Keep all checkboxes as the same status, all the check statuses of them are both true, or just both false.

Reply:
makes sense